### PR TITLE
Add a HTTP request module with authorization

### PR DESCRIPTION
**Pull request by cspiel1**
Provides the HTTP request feature of https://github.com/baresip/re/pull/33 in form of a new baresip module "httpreq".
- Makes use of struct network which also was moved to libre.
- GET, POST requests
- TLS
- basic, digest and bearer authentication

PR https://github.com/baresip/re/pull/34 adds struct network to libre.

Travis will be happy if https://github.com/baresip/re/pull/33 and https://github.com/baresip/re/pull/34 are merged.

### CODE REVIEW

**Comment by oej** [comment]
You only have a setting for client cert. I would like to see a setting for client key to have them in separate files.

**Comment by cspiel1** [comment]
> You only have a setting for client cert. I would like to see a setting for client key to have them in separate files.

Okay, makes sense!

**Comment by alfredh** [comment]
what is the purpose of this
new module?

**Comment by cspiel1** [comment]
> what is the purpose of this
> new module?

Allows to interact with a remote REST interface.

File: include/baresip.h [hunk 1]
```diff
@@ -333,17 +333,6 @@ struct config_avt {
...
-struct config_net {
-	int af;                 /**< AF_UNSPEC, AF_INET or AF_INET6 */
-	char ifname[64];        /**< Bind to interface (optional)   */
-	struct {
-		char addr[64];
-		bool fallback;
-	} nsv[NET_MAX_NS];      /**< Configured DNS nameservers     */
-	size_t nsc;             /**< Number of DNS nameservers      */
-};
-
```
Comment (alfredh) [inline include/baresip.h:346]: I dont agree with this change.

This code belongs to baresip. if you need similar code in a different project, it is better
to write the code from scratch or to copy parts of it.

Comment (cspiel1) [reply include/baresip.h:346]: Do you mean you don't want to have "struct network" in libre? I could revert this commit and leave network in baresip.
Or do you mean you don't find the module useful?
Or do you mean that you don't find request.c in libre useful at all?

**Comment by alfredh** [comment]
I dont think that we should move `struct config_net` to re.

the module itself is useful.

the patch itself is quite large, it would be easier to review if the patch was smaller...

**Comment by cspiel1** [comment]
I removed the network commit.
I will try to split up the large commit "httpreq: add HTTP request module" into handsome chunks.

**Comment by cspiel1** [comment]
Thanks Alfred for review and merge!

**Pull request merged**